samples/boards/stm32/power_mgmt/stop3/src/main.c

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) 2021 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/sys/printk.h>
+#include <stm32u5xx_ll_pwr.h>
+
+#define SLEEP_TIME_MS   2000
+
+static const struct gpio_dt_spec led =
+	GPIO_DT_SPEC_GET(DT_ALIAS(led0), gpios);
+
+int main(void)
+{
+	bool led_is_on = true;
+
+	__ASSERT_NO_MSG(gpio_is_ready_dt(&led));
+
+	printk("Device ready\n");
+	gpio_pin_configure_dt(&led, GPIO_OUTPUT_ACTIVE);
+
+	while (true) {
+		gpio_pin_set(led.port, led.pin, (int)led_is_on);
+		/* In STOP3, GPIOs are disabled. Only pull-up/pull-down can be enabled.
+		 * So we enable pull-up/pull-down based on LED status
+		 */
+		if (led_is_on) {
+			PWR->PDCRB &= ~(uint32_t)(1 << 7);
+			PWR->PUCRB |= (1 << 7);
+		} else {
+			PWR->PUCRB &= ~(uint32_t)(1 << 7);
+			PWR->PDCRB |= (1 << 7);
```

Comment:
Would be ideal to make it configurable from led.port and led.pin.
Doesn't have to be much complex, maybe using an array to get the address?
If too complex in short term, let's at minimum add a comment stating this is expecting such led value (or generating error is led value is not as expected)

Reply:
I used the code proposed by @ajarmouni-st and also compute the parameter from led.port.